server/src/controllers/auth.ts

```diff
@@ -72,43 +80,111 @@ const login = async (req: Request, res: Response) => {
+    // If password doesn't match
+    if (!matchPassword) {
+      return res.status(401).json({ message: "Incorrect password" });
+    }
+
+    // Create a jwt token
+    const token = generateToken({
+      _id: emailExists._id,
+      email: emailExists.email,
+    });
+
+    return res.status(200).json({
+      data: { token: token },
+      message: "User logged-in successfully",
+    });
+  } catch (error) {
+    console.error(error);
+    return res.status(500).json({ message: "Internal server error" });
+  }
+};
+
+/*
+1. Take user email
+2. Validate email
+3. Generate a token with 1-hour expiry
+4. Send token to user's email
+*/
+const forgotPassword = async (req: Request, res: Response) => {
+  const { email } = req.body;
+
+  try {
+    if (!email) {
+      return res.status(400).json({ message: "Email is required" });
+    } 
+    if (!isValidEmail(email)) {
       return res.status(400).json({ message: "Invalid email address" });
-    } else {
-      const emailExists = await getDetails("email", email);
-
-      // Check if user registered or not
-      if (!emailExists) {
-        return res.status(404).json({ message: "Email not registered" });
-      } else {
-        // Compare the password saved in DB and entered by user.
-        const matchPassword = await bcrypt.compare(
-          password,
-          emailExists.password
-        );
-
-        // If password doesn't match
-        if (!matchPassword) {
-          return res.status(401).json({ message: "Incorrect password" });
-        } else {
-          // Create a jwt token
-          const token = generateToken({
-            _id: emailExists._id,
-            email: emailExists.email,
-          });
-
-          return res.status(200).json({
-            data: { token: token },
-            message: "User logged-in successfully",
-          });
-        }
-      }
     }
+
+    const user = await getDetails("email", email);
+    if (!user) {
+      return res.status(404).json({ message: "Email not registered" });
+    }
+
+    const token = generateToken({ _id: user._id, email: user.email }, '1h'); // Token expires in 1 hour
+    const resetLink = `${process.env.CLIENT_URL}/reset-password?token=${token}`;
+
+    // Send email
+    await transporter.sendMail({
+      to: email,
+      subject: 'Password Reset Request',
+      text: `Click the following link to reset your password: ${resetLink}`,
+      html: `<p>Click the following link to reset your password: <a href="${resetLink}">${resetLink}</a></p>`,
+    });
+
+    return res.status(200).json({ message: "Password reset link sent" });
+  } catch (error) {
+    console.error(error);
+    return res.status(500).json({ message: "Internal server error" });
+  }
+};
+
+/*
+1. Take token and new password
+2. Validate token
+3. Hash new password and update in DB
+4. Return success message
+*/
+const resetPassword = async (req: Request, res: Response) => {
+  const { token, password } = req.body;
```

Comment:
As mentioned, token will be verified by middleware function. So from client just send in headers i mentioned you can check services folder of client.